lib/Dialect/Handshake/HandshakeTypes.cpp

```diff
@@ -115,58 +262,41 @@ static Type parseChannelAfterLess(AsmParser &odsParser) {
                         "which is to be a Type");
     return nullptr;
   }
+
+  // TODO: This check is also called in verify. We should consider refactoring
+  // this.
```

Comment:
If I remember things correctly we have to verify things here as well otherwise we will crash on an invalid IR.